_docs/ux-design.md

```diff
@@ -0,0 +1,383 @@
+[instrumentation packages](#instrumentation-packages) definitions file.
+
+### Clean-Room Usage
+
+Some users want to be able to apply compile-time instrumentation to a codebase
+without many _any_ modification to it.
+
+To support these users, the `otel` tool can be used without making any
+persistent modifications to the codebase: when using `otel go build` without
+having created any configuration, the tool will automatically analyze the
+project's dependency tree to self-configure and build the project, before
+cleaning up any file system changes that may have needed to be made in the
+process.
+
+The self-configuration can be influenced by passing any and all relevant build
+flags to the `otel` command as part of the build.
+
+It is important to note that this mode of operation does not produce
+reproductible builds. Users who want or need fully reproductible builds must use
+the explicit set-up procedure.
+
+### Uninstalling
+
+Removing auto-instrumentation configuration is as simple as removing the related
+tool dependencies from `go.mod` and removing the `otel.instrumentation.go`
+file.
+
+## Instrumentation Packages
+
+A majority of users of the OpenTelemetry compile-time instrumentation tool will
+rely on instrumentation packages to instrument their application. These are
+standard Go packages that are part of a Go module and contain either (or both):
+
+- a `otel.instrumentation.yml` file that declares all instrumentation
+  configuration that is vended by this package (using the schema defined in the
+  next section);
+- a `otel.instrumentation.go` file that imports at least one valid
+  instrumentation package:
+  ```go
+  //go:build tools
+  package tools
+
+  import (
+  	_ "github.com/open-telemetry/opentelemetry-go/db"
+  	_ "github.com/open-telemetry/opentelemetry-go/http"
+  )
+  ```
+
+### Schema
+
+The schema of the `otel.instrumentation.yml` file (as well as `.otel.yml`
+files, which are used by project-specific instrumentation settings) is described
+by the following document:
+
+```yml
+%YAML 1.2
+---
+# yaml-language-server: $schema=https://go-otel.opentelemetry.io/schemas/instrumentation
+
+meta: # An optional block of metadata about the configuration file
+   description: # Optional
+      |-
+         A description of what this configuration does, intended to inform
+         end-users about this instrumentation package.
+   caveats: # Optional
+      - |-
+            An array of strings detailing caveats from using this
+            instrumentation package. These may be presented to the users when
+            they install this package for the first time.
+
+instrumentation: # Required with at least 1 item
+   foo: # A unique identifier for this instrumentation item within this file
+      description: #Optional
+         |-
+            A decription of this instrumentation configuration, intended for
+            end-users.
+      pointcut: # Required
```

Comment:
I overlooked an important point: regarding this pointcut, I believe we should support Go source code rather than AST node manipulation.